core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -328,6 +328,14 @@ object SparkSubmit {
       }
     }
 
+    // In yarn mode for a python app, if PYSPARK_ARCHIVES_PATH is in the user environment
+    // add pyspark archives to files that can be distributed with the job
+    if (args.isPython && clusterManager == YARN){
+      sys.env.get("PYSPARK_ARCHIVES_PATH").map { archives =>
```

Comment:
spark administrator  can set  pyspark's zip of local or hdfs path to  PYSPARK_ARCHIVES_PATH in spark-env.sh.  and then user run python application as before.